### PR TITLE
windows: utf8 documents, prompt for update, refactor

### LINKED ISSUE #277: Windows client should tell the user when they need to update
if core ever sends back a `ClientUpdateRequired`, show in the status area that an `Update is required`.

 The following enums now indicate that a `ClientUpdateRequired`:
+ `CreateAccountError`
+ `ImportAccountError`
+ `CalculateWorkError`
+ `ExecuteWorkError`

### LINKED ISSUE #319: é crashes the windows client
+ Should core be thinking and dealing in terms of bytes?
+ Is encoding between client A, core, and other scenarios well thought out and correct?
+ There are some weird interactions between windows and ipad

### PR DESCRIPTION
**Pull request by tvanderstad**
Closes #319
Closes #277

### CODE REVIEW

File: clients/windows/lockbook/Settings.xaml.cs [hunk 1]
```diff
@@ -6,26 +6,78 @@
...
 
 namespace lockbook {
 
     public sealed partial class SignInContentDialog : ContentDialog {
         const ulong BYTE = 1;
-        const ulong KILOBYTES = BYTE* 1000;
-        const ulong MEGABYTES = KILOBYTES* 1000;
-        const ulong GIGABYTES = MEGABYTES* 1000;
-        const ulong TERABYTES = GIGABYTES* 1000;
+        const ulong KILOBYTES = BYTE * 1000;
```
Comment (Parth) [inline clients/windows/lockbook/Settings.xaml.cs:16]: https://github.com/lockbook/lockbook/pull/502

should probably use this, hot off the press

File: clients/windows/lockbook/App.xaml.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using System.Threading.Tasks;
```
Comment (Parth) [inline clients/windows/lockbook/App.xaml.cs:4]: This commit message: `dont exit app when clearing state, auto clear state when needed` doesn't seem to be what actually happens -- we don't actually nuke a user's state on `StateRequiresCleaning` do we? If so we should not.

File: clients/windows/lockbook/SignUp.xaml [hunk 3]
```diff
@@ -128,7 +128,12 @@
                     <TextBox
                         Width="300"
                         x:Name="usernameTextBox"
-                        Header="Select a Username" />
+                        Header="Username" />
+                    <TextBox
```
Comment (Parth) [inline clients/windows/lockbook/SignUp.xaml:132]: Not on board with all users seeing this, it should be behind some debug config or something

**Review by Parth — CHANGES_REQUESTED** [review]
Left some comments in the commits, also this closes several issues, you should add closes statements. I'll throw this on a device and QA it when you're ready

**Comment by tvanderstad** [comment]
closed in favor of https://github.com/lockbook/lockbook/pull/551 due to a mysterious yet tragic git accident

**Pull request closed without merging**